File: Api/GamContentSystemApi/GamContentSystemApi/Database/Entities/Quest.cs
namespace GamContentSystemApi.Database.Entities
{
    public class Quest : BaseEntity
    {
        public string? Title { get; set; }
        public int? StoryChapterId { get; set; }
        public int? QuestActionId { get; set; }
        public int? BaseItemId { get; set; }

        public StoryChapter? StoryChapter { get; set; }
        public QuestAction? QuestAction { get; set; }
        public BaseItem? BaseItem { get; set; }
    }
}

File: Ui/src/interfaces/quest.ts
import { BaseItem } from "./item";
import { QuestAction } from "./questAction";
import { StoryChapter } from "./storyChapter";

export interface Quest {
  id: number;
  title: string;
  storyChapter: StoryChapter;
  questAction?: QuestAction;
  baseItem?: BaseItem;
}
